chain/types/ethtypes/eth_legacy_homestead_transactions.go

```diff
@@ -0,0 +1,368 @@
+package ethtypes
+
+import (
+	"fmt"
+
+	"golang.org/x/crypto/sha3"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	gocrypto "github.com/filecoin-project/go-crypto"
+	"github.com/filecoin-project/go-state-types/big"
+	typescrypto "github.com/filecoin-project/go-state-types/crypto"
+
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+const (
+	legacyHomesteadTxSignatureLen = 66
+)
+
+var _ EthTransaction = (*EthLegacyHomesteadTxArgs)(nil)
+
+type EthLegacyHomesteadTxArgs struct {
+	Nonce    int         `json:"nonce"`
+	GasPrice big.Int     `json:"gasPrice"`
+	GasLimit int         `json:"gasLimit"`
+	To       *EthAddress `json:"to"`
+	Value    big.Int     `json:"value"`
+	Input    []byte      `json:"input"`
+	V        big.Int     `json:"v"`
+	R        big.Int     `json:"r"`
+	S        big.Int     `json:"s"`
+}
+
+func (tx *EthLegacyHomesteadTxArgs) ToEthTx(smsg *types.SignedMessage) (EthTx, error) {
+	from, err := EthAddressFromFilecoinAddress(smsg.Message.From)
+	if err != nil {
+		// This should be impossible as we've already asserted that we have an EthAddress
+		// sender...
+		return EthTx{}, fmt.Errorf("sender was not an eth account")
+	}
+	hash, err := tx.TxHash()
+	if err != nil {
+		return EthTx{}, fmt.Errorf("failed to get tx hash: %w", err)
+	}
+
+	gasPrice := EthBigInt(tx.GasPrice)
+	ethTx := EthTx{
+		ChainID:  ethLegacyHomesteadTxChainID,
+		Type:     ethLegacyHomesteadTxType,
+		Nonce:    EthUint64(tx.Nonce),
+		Hash:     hash,
+		To:       tx.To,
+		Value:    EthBigInt(tx.Value),
+		Input:    tx.Input,
+		Gas:      EthUint64(tx.GasLimit),
+		GasPrice: &gasPrice,
+		From:     from,
+		R:        EthBigInt(tx.R),
+		S:        EthBigInt(tx.S),
+		V:        EthBigInt(tx.V),
+	}
+
+	return ethTx, nil
+}
+
+func (tx *EthLegacyHomesteadTxArgs) ToUnsignedFilecoinMessage(from address.Address) (*types.Message, error) {
+	mi, err := getFilecoinMethodInfo(tx.To, tx.Input)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get method info: %w", err)
+	}
+
+	return &types.Message{
+		Version:    0,
+		To:         mi.to,
+		From:       from,
+		Nonce:      uint64(tx.Nonce),
+		Value:      tx.Value,
+		GasLimit:   int64(tx.GasLimit),
+		GasFeeCap:  tx.GasPrice,
+		GasPremium: tx.GasPrice,
+		Method:     mi.method,
+		Params:     mi.params,
+	}, nil
+}
+
+func (tx *EthLegacyHomesteadTxArgs) ToVerifiableSignature(sig []byte) ([]byte, error) {
+	if len(sig) != legacyHomesteadTxSignatureLen {
+		return nil, fmt.Errorf("signature should be %d bytes long, but got %d bytes", legacyHomesteadTxSignatureLen, len(sig))
```

Comment:
Not sure but might be instructive to add a little more information to avoid confusion with the literal length of eth homestead signatures being 1 byte less.
```suggestion
		return nil, fmt.Errorf("signature should be %d bytes long (1 byte metadata, %d bytes sig data), but got %d bytes", legacyHomesteadTxSignatureLen, legacyHomesteadTxSignatureLen - 1, len(sig))
```

Reply:
Done.